src/client.cpp

```diff
@@ -2199,13 +2199,13 @@ void Client::sendChangePassword(const std::wstring &oldpassword,
 }
 
 
-void Client::sendDamage(u8 damage)
+void Client::sendDamage(s8 damage)
 {
 	DSTACK(__FUNCTION_NAME);
 	std::ostringstream os(std::ios_base::binary);
 
 	writeU16(os, TOSERVER_DAMAGE);
-	writeU8(os, damage);
+	writeS8(os, damage);
```

Comment:
I haven't looked at this closely but my gut feeling is that you should leave it as u8 and cast it after being read (or cast to u8 before being written).